### PR TITLE
#478 fix for invalid types

### PR DESCRIPTION
**Pull request by amcdnl**

### CODE REVIEW

File: src/provider.tsx [hunk 1]
```diff
@@ -31,7 +31,8 @@ export const IntercomProvider: React.FC<React.PropsWithChildren<
   const isBooted = React.useRef(false);
   const isInitialized = React.useRef(false);
 
-  if (!isEmptyObject(rest) && __DEV__)
+  const restKeys = Object.keys(rest).filter(k => k.startsWith('data-'));
```
Comment (devrnt) [inline src/provider.tsx:34]: This predicate should be the opposite

**Comment by devrnt** [comment]
Thanks for this PR! Did some changes, would you like to take a look?

**Comment by devrnt** [comment]
**Size Change:** +71 B (+1%) 

**Total Size:** 9.46 kB

| Filename | Size | Change |
| :--- | :---: | :---: |
| `dist/react-use-intercom.cjs.development.js` | 3.66 kB | +30 B (+1%) |
| `dist/react-use-intercom.cjs.production.min.js` | 1.99 kB | -5 B (0%) |
| `dist/react-use-intercom.esm.js` | 3.65 kB | +46 B (+1%) |

<details><summary>ℹ️ <strong>View Unchanged</strong></summary>

| Filename | Size |
| :--- | :---: |
| `dist/index.js` | 152 B |

</details>

**Review by amcdnl — COMMENTED** [review]
Looks great!

**Pull request merged**